### DISCUSSION TITLE
Failed to run sql query: column "id" specified in USING clause does not exist in left table; but it definetly does exist.

### DISCUSSION THREAD

**Discussion by mckinlde** [supabase/supabase#16176] [Questions]
I am trying to run a sql query in sql editor to define a function:
`create function comment_tree(
  comment_id int
) returns json as $$
  select json_build_object(
    'comment_id', c.comment_id,
    'user', json_build_object(
      'username', profiles.username,
      'user_id', profiles.id
    ),
    'body', body,
    'created_at', c.created_at,
    'children', children
  )
  from comments c
    left join profiles using(id),
    lateral (
      select
        coalesce(json_agg(
          comment_tree(comments.comment_id)
          order by created_at asc
        ), '[]') as children
      from comments
      where parent_id = c.comment_id
    ) as get_children
  where
    c.comment_id = comment_tree.comment_id
$$ language sql stable;`

and recieve the err:
`Failed to run sql query: column "id" specified in USING clause does not exist in left table`

but the column does exist, and to prove that I run the query
`select id from profiles`
which returns values as expected.

This seems like a bug behind the sql editor; can anyone help?

**Comment by GaryAustin1** [comment]
I believe that is saying comments does not have an id column.  Using wants `id` in both comments and profiles to match up.
